_upgrading/sopel-8.md

```diff
@@ -0,0 +1,420 @@
+---
+title: Sopel 8.0 Migration
+covers_from: 7.x
+covers_to: 8.0
+---
+
+# Sopel 8.0 Migration
+
+Version 8.0 brings Sopel into the modern era of Python, finally dropping support
+for Python 2 and all end-of-life Python 3 releases as of December 2023. This let
+us make significant updates under the hood to future-proof things like the IRC
+connection backend and event system.
+
+## Owner/admin usage changes
+
+### Updated Python requirements & support policy
+
+Sopel 8.0 requires Python 3.8 or higher. Support for Python 2.7 and 3.3–3.7 has
+been removed. In exchange, known issues sometimes preventing use of Sopel 7 with
+Python 3.11+ have been fixed.
+
+During the lifecycle of Sopel 8.x, we will add new Python releases to our test
+suite as soon as possible. We will keep testing against end-of-life Python
+versions unless and until it becomes technically impossible to do so. However,
+if testing against an EOL Python version does become impossible, we will drop it
+in the next maintenance release of Sopel.
```

Comment:
No one gave me feedback on this part; I just thought it could be worded better while looking at the other comments around it.

```suggestion
versions unless and until it becomes technically impossible to do so—in which
case Sopel will officially drop support for that EOL Python release in its next
maintenance release.
```